generators/paths/domain_parser/code_snippet_common.md

```diff
@@ -0,0 +1,36 @@
+```python
+import re
+
+def domain_parser(link:str, include_sub_domain:bool = False) -> str:
+    """
+    @param link: the input link
+    @param include_sub_domain: Attribute whether subdomain in front of domain is desired or not
+    @return: rootdomain
+    """
+
+    clean_link = re.sub("www.", "",link)
+    parts = clean_link.split("/")
+    domain = parts[2]
+    if include_sub_domain == False:
```

Comment:
The idea was to exclude the subdomain after the root domain. Here on the other hand, I give the opportunity to in- or exclude the subdomain before the root-domain, e.g. `app.dev.kern.ai `as `kern.ai` without the subdomain and `app.dev.kern.ai` with the subdomain on the front. Same in `mail.google.com`. you would get google.com or mail.google.com

Reply:
It actually was just an ideal, because I was unsure how to read the task (it sounded a bit ambigous to me): Is in the end just google.com requiered or also mail.google.com. So I gave both possibilities.